src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -35,7 +35,7 @@ public RemoteWalkContext(SourceCacheContext cacheContext, PackageNamespacesConfi
         public ILogger Logger { get; }
         public IList<IDependencyProvider> ProjectLibraryProviders { get; }
         public IList<IRemoteDependencyProvider> LocalLibraryProviders { get; }
-        public IList<IRemoteDependencyProvider> RemoteLibraryProviders { get; }
+        private IReadOnlyList<IRemoteDependencyProvider> RemoteLibraryProviders { get; }
```

Comment:
I feel like this is over complicating things. 

I know the current API is not ideal, but consistency has a certain appeal as well.

Reply:
Reverted this change as per offline conversation.